nixos/modules/services/misc/autosuspend.nix

```diff
@@ -0,0 +1,238 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.services.autosuspend;
+
+  settingsFormat = pkgs.formats.ini {};
+
+  checks =
+    mapAttrs'
+    (n: v: {
+      name = "check.${n}";
+      value = filterAttrs (_: v: v != null) v;
+    })
+    cfg.checks;
+  wakeups =
+    mapAttrs'
+    (n: v: {
+      name = "wakeup.${n}";
+      value = filterAttrs (_: v: v != null) v;
+    })
+    cfg.wakeups;
+
+  # Whether the `Smb` check is enabled, and therefore requires `smbstatus` in path
+  needsSamba =
+    (filterAttrs (n: v:
+      (
+        if v.class == null
+        then n
+        else v.class
+      )
+      == "Smb"
+      && v.enabled)
+    cfg.checks)
+    != {};
+
+  autosuspend-conf =
+    settingsFormat.generate "autosuspend.conf" ({general = cfg.settings;} // checks // wakeups);
+
+  autosuspend = cfg.package;
+
+  version = "4.3.0";
+
+  checkType = with types;
+    submodule {
+      freeformType = settingsFormat.type.nestedTypes.elemType;
+
+      options.enabled = mkEnableOption (mdDoc "this activity check");
+
+      options.class = mkOption {
+        default = null;
+        type = nullOr (enum [
+          "ActiveCalendarEvent"
+          "ActiveConnection"
+          "ExternalCommand"
+          "JsonPath"
+          "Kodi"
+          "KodiIdleTime"
```

Comment:
Seems like Kodi and MPD use Python libraries to do the checks.  However, I can't ensure that they work, because it seems `autosuspend` on nixpkgs was built without the required Python libraries :/ So I'd need to patch that first.  In the meantime, I'll add requirements needed by `xidletime`.

Reply:
Yeah that's basically the approach used by upstream `autosuspend`.  The default is built without any extra dependencies, and you can pick from a list of extra features to enable, which are required to do some of the checks like Kodi.